src/hooks/useSubscriptions.js

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import {
+  transformProposalData,
+  transformStakeHistoryData,
+} from '../lib/data-utils'
+
+export function useProposalsSubscription(convictionVoting) {
+  const [proposals, setProposals] = useState([])
+
+  const proposalsSubscription = useRef(null)
+
+  const onProposalsHandler = useCallback((proposals = []) => {
+    const transformedProposals = proposals.map(transformProposalData)
+    setProposals(transformedProposals)
+  }, [])
+
+  useEffect(() => {
+    if (!convictionVoting) {
+      return
+    }
+
+    proposalsSubscription.current = convictionVoting.onProposals(
+      onProposalsHandler
+    )
+
+    return () => proposalsSubscription.current[0]()
```

Comment:
The  `onPropsals` function is supposed to return the object with the `unsubscribe` function (see https://github.com/1Hive/conviction-voting-app/blob/bdd20ffa7110475d15eb36576fa727b3ba5c6e0a/connector/src/entities/ConvictionVoting.ts#L24) 
but it's actually returning an array where the first element is the unsubscribe function